datalad_catalog/webcatalog.py

```diff
@@ -151,52 +204,61 @@ def set_main_dataset(self):
             json.dump(main_obj, f)
         return main_file
 
-    def get_config_source(self, source_str=None):
+    def get_config_source(
+        self, source_str: str = None, config_level: str = "catalog"
+    ):
         """"""
-        # If no source_str provided, determine
-        if self.is_created():
-            # If catalog already exists, return config if it exists, otherwise None
-            config_path = Path(self.location / "config.json")
-            if config_path.exists():
-                return config_path
-            # TODO: if catalog exists without config file, should one be created from default?
-            return None
-        else:
-            # If catalog does not exist, return config if specified, otherwise default
+        if config_level == "catalog":
+            if self.is_created():
+                # If catalog already exists, return config if it exists, otherwise default
+                catalog_config_path = Path(self.location / "config.json")
+                if catalog_config_path.exists():
+                    return catalog_config_path
+                return Path(self.default_config_dir / "config.json")
+            else:
+                # If catalog does not exist, return config if specified, otherwise default
+                if source_str is not None:
+                    return Path(source_str)
+                else:
+                    return Path(self.default_config_dir / "config.json")
+        if config_level == "dataset":
             if source_str is not None:
                 return Path(source_str)
             else:
-                return Path(self.config_dir / "config.json")
+                # Do not return default here, this will be done on the Node or MetaItem class,
+                # where dataset id and version are available.
+                return None
 
-    def get_config(self):
+    def get_config(self, config_level: str = "catalog"):
         """"""
         # Read metadata from file
-        with open(self.config_path) as f:
-            if self.config_path.suffix == ".json":
-                return json.load(f)
-            if self.config_path.suffix in [".yml", ".yaml"]:
-                return yaml.safe_load(f)
+        cfg_paths = {
+            "catalog": self.catalog_config_path,
+            "dataset": self.dataset_config_path,
+        }
+        config_path = cfg_paths[config_level]
+        return load_config_file(config_path)
 
     def write_config(self, force=False):
         """"""
-
         # Copy content specified by config
         if (
-            cnst.LOGO_PATH in self.config
-            and self.config[cnst.LOGO_PATH]
-            and self.config[cnst.LOGO_PATH] != "artwork/catalog_logo.svg"
+            cnst.LOGO_PATH in self.catalog_config
+            and self.catalog_config[cnst.LOGO_PATH]
+            and self.catalog_config[cnst.LOGO_PATH]
+            != "artwork/catalog_logo.svg"
```

Comment:
This can probably be simplified as:
```suggestion
            self.config.get(cnst.LOGO_PATH) not in (None, "artwork/catalog_logo.svg")
```